config/nrfconnect/chip-module/Kconfig

```diff
@@ -70,3 +70,145 @@ config CHIP_DEBUG_SYMBOLS
 	default y
 	help
 	  Build the application with debug symbols.
+
+# Factory data
+config CHIP_FACTORY_DATA_ENABLE
+	bool "Enables Factory Data support"
+	default n
+	help
+		Enables generating a factory data after build.
+		To generate factory data the nrfconnect_generate_factory_data function from
+		generate_factory_data.cmake should be called.
+		In result a new output files will be created:
+		- merged_fd.hex file containing firmware and factory data merged
+		- factory_data.hex file containing only a factory data with proper memory offset.
+  
+config CHIP_FACTORY_DATA_PARTITION
+	hex "Localization in NVS storage where Factory Data partition will be stored"
+	depends on CHIP_FACTORY_DATA_ENABLE
+	default 0xfb000
+	help 
+		FLASH Memory location where created factory data will be stored.
+
+config CHIP_FACTORY_DATA_PARTITION_SIZE
+	hex "Maximum size of partition data"
+	depends on CHIP_FACTORY_DATA_ENABLE
+	default 0x1000
+	help
+		Provide maximum partition size to verify that 
+		new data fit to a place located in nvs memory
+
+config CHIP_FACTORY_DATA_GENERATE_DAC_CERT
+	bool "Enables generation DAC cert after build"
+	default n
+	help
+		Enables generation Device Attestation Certificate according to given Vendor and Product IDs
+		To generate DAC Cert a chip-cert program must be available in system variables environment.
+		Currently only Linux system is supported to generate DAC cert
+
+config CHIP_FACTORY_DATA_GENERATE_PAI_CERT
+	bool "Enables generation PAI cert after build"
+	help
+		Enables generation Product Attestation Intermediate Certificate according to given Vendor and Product IDs
+		To generate PAI Cert a chip-cert programmust be available in system variables environment.
+		Currently only Linux system is supported to generate PAI cert
+
+config CHIP_FACTORY_DATA_GENERATE_SPAKE2_VERIFIER
+	bool "Enables generation spake2 verifier"
+	depends on CHIP_FACTORY_DATA_ENABLE
+	default n
+	help
+		Enables generation spake2 verifier according to given iteration counter, salt and passcode
+		To generate Spake2 verifier a spake2p program must be available in system variables environment.
+		Currently only Linux system is supported to generate PAI cert
+
+config CHIP_FACTORY_DATA_USE_DEFAULTS_CERTS_PATH
+	bool "Uses a default certificates located in Matter repository"
+	depends on CHIP_FACTORY_DATA_ENABLE
+	default n
+	help
+		Uses a default certificates, which are located in Matter repository
+		in credentials/development/attestation/ directory 
+
+# Factory data parameters
+config CHIP_DEVICE_SERIAL_NUMBER
+	string "Serial number of device"
+	default "1010560536528535133"
+	help
+		Provide serial number of device to store in nvs memory
```

Comment:
Some general comments to many places, so please apply everywhere:
* Add dots at the end of descriptions in help.
* Replace nvs memory with flash memory.
* Remove "after build" mentions, I don't think it is important for user to know.